lib/ansible/template/vars.py

```diff
@@ -57,6 +57,22 @@ def __init__(self, templar, globals, locals=None, *extras):
                     elif key not in ('context', 'environment', 'template'):
                         self._locals[key] = val
 
+    def __iter__(self):
+        for k in self._templar._available_variables:
```

Comment:
Reason being this [code from jinja](https://github.com/pallets/jinja/blob/d67f0fd4cc2a4af08f51f4466150d49da7798729/jinja2/runtime.py#L181):
```python
return dict(self.parent, **self.vars)
```
I'm assuming that it intends for self.parent to use the dict `mapping` constructor, and inheriting from collections.Mapping gives it the hint that it needs here:
```
class dict(object)
 |  dict() -> new empty dictionary
 |  dict(mapping) -> new dictionary initialized from a mapping object's
 |      (key, value) pairs
 |  dict(iterable) -> new dictionary initialized as if via:
 |      d = {}
 |      for k, v in iterable:
 |          d[k] = v
```